Sources/Fluent/Database/ThreadConnectionPool.swift

```diff
@@ -0,0 +1,75 @@
+import Foundation
+
+/**
+    Responsible for maintaing of pool
+    of connections, one for each thread.
+ 
+    Uses a Driver to make new connections.
+*/
+public final class ThreadConnectionPool {
+    var makeConnection: ConnectionFactory
+    var storage: [pthread_t: (conn: Connection, time: Int)]
```

Comment:
Unless I'm mistaken, the point isn't for other threads to reuse the connection, it's for a single thread to reuse its connection (required by Transactions). And the only way for a series of calls to use the same connection but lock it is either use thread ids or change the API to pass the connection around.